packages/webapp/pages/account/invite.tsx

```diff
@@ -0,0 +1,125 @@
+import { SocialShareList } from '@dailydotdev/shared/src/components/widgets/SocialShareList';
+import { useCopyLink } from '@dailydotdev/shared/src/hooks/useCopy';
+import AccountContentSection from '../../components/layouts/AccountLayout/AccountContentSection';
+import { AccountPageContainer } from '../../components/layouts/AccountLayout/AccountPageContainer';
+import { getAccountLayout } from '../../components/layouts/AccountLayout';
+
+const AccountInvitePage = (): ReactElement => {
+  const { user } = useAuthContext();
+  const container = useRef<HTMLDivElement>();
+  const referredKey = generateQueryKey(RequestKey.ReferredUsers, user);
+  const { url, referredUsersCount } = useReferralCampaign({
+    campaignKey: ReferralCampaignKey.Generic,
+  });
+  const inviteLink = url || link.referral.defaultUrl;
+  const [copyingLink, onCopyLink] = useCopyLink(() => inviteLink);
+  const usersResult = useInfiniteQuery<ReferredUsersData>(
+    referredKey,
+    ({ pageParam }) =>
+      request(graphqlUrl, REFERRED_USERS_QUERY, {
+        after: typeof pageParam === 'string' ? pageParam : undefined,
+      }),
+    {
+      getNextPageParam: (lastPage) =>
+        lastPage?.referredUsers?.pageInfo?.hasNextPage &&
+        lastPage?.referredUsers?.pageInfo?.endCursor,
+    },
+  );
+
+  const onNativeShare = async () => {
+    await navigator.share({
+      title: labels.referral.generic.inviteText,
+      url: inviteLink,
+    });
+  };
+
+  const onTrackShare = () => {
+    // add tracking here
+  };
+
+  return (
+    <AccountPageContainer title="Invite friends">
+      <TextField
+        name="inviteURL"
+        inputId="inviteURL"
+        label="Your unique invite URL"
+        type="url"
+        autoComplete="off"
+        value={inviteLink}
+        fieldType="tertiary"
+        actionButton={
+          <Button
+            buttonSize={ButtonSize.Small}
+            className="btn-primary"
+            onClick={() => onCopyLink()}
+          >
+            {copyingLink ? 'Copying...' : 'Copy link'}
+          </Button>
+        }
+        readOnly
+      />
+      <span className="p-0.5 my-4 font-bold typo-callout text-theme-label-tertiary">
+        or invite with
+      </span>
+      <div className="flex flex-row flex-wrap gap-4">
+        <SocialShareList
+          link={inviteLink}
+          description={labels.referral.generic.inviteText}
+          onNativeShare={onNativeShare}
+          onClickSocial={onTrackShare}
+        />
+      </div>
+      <AccountContentSection
+        title="Referred"
+        description="Here you will see all your active members who joined to daily.dev. Who knows? maybe someday it will be worth something 😉"
+      >
+        <UserList
+          users={usersResult.data?.pages.reduce((acc, p) => {
+            p?.referredUsers.edges.forEach(({ node }) => {
+              acc.push(node);
+            });
+
+            return acc;
```

Comment:
Should we not do this reduce outside the render?

Reply:
Yep. Was actually thinking of creating a hook here, seems to be reused in numerous places.